### ISSUE TITLE
Support for DDD enumeration classes

### ISSUE THREAD

**Issue by FaizulHussain** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#694]
Instead of generating standard enums, will it be possible to generate enumeration classes as discussed here:

https://docs.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/enumeration-classes-over-enum-types

**Comment by sjh37** [comment]
I could certainly generate it, however it would not add benefit unless it was a partial class to which you could add your own methods.

**Comment by FaizulHussain** [comment]
you are right, it would have little value. i guess the work around is to generate the partial class instead of enums, and then use DDD pattern on another partial class. this works for me

thanks

**Comment by sjh37** [comment]
## Not Feasible / Out of Scope

As discussed, generating DDD enumeration classes without making them `partial` provides little benefit over the existing enum generation. You can use the existing enum output and extend it in a partial class with your DDD methods. Closing — if there is significant demand for this feature, please upvote by adding a 👍 reaction.
